driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -72,7 +76,12 @@ public static IntegerExpression of(final long of) {
     public static NumberExpression of(final double of) {
         return new MqlExpression<>((codecRegistry) -> new AstPlaceholder(new BsonDouble(of)));
     }
+    public static NumberExpression of(final BigDecimal of) {
```

Comment:
BigDecimal is fine, but we need Decimal128 too, as not all Decimal128 values are representable as BigDecimal.

Reply:
I'm having second thoughts about the inclusion of this method.  Given that Decimal128 use is fairly specialized, forcing the application to call `Decimal128.of(BigDecimal)` doesn't seem like too much of an imposition, and makes it clear that we don't actually support all BigDecimal instances, but only the ones that can be represented as Decimal128.

Reply:
Agreed